### PR TITLE
Classification improvements

### PR DESCRIPTION
**Pull request by sovrasov**
- Several bugs are fixed
- Models were refactored a bit
- Add models from pytorchcv (for classification with softmax only)
- Add an analog of ImageFolder dataset from torchvision

### CODE REVIEW

**Comment by sovrasov** [comment]
@evgeny-izutov please take a look

**Review by evgeny-izutov — APPROVED** [review]
